src/generic/stage2_build_tape.h

```diff
@@ -293,13 +376,15 @@ unified_machine(const uint8_t *buf, size_t len, ParsedJson &pj) {
     /* we found an object inside an object, so we need to increment the
      * depth                                                             */
     depth++;
-    if (depth >= pj.depth_capacity) {
-      goto fail;
+    if(depth < pj.depth_capacity) {
```

Comment:
Hmm, is this an attempt to improve default branch prediction because it should be a backwards jump?